draft-ietf-gnap-core-protocol.md

```diff
@@ -872,6 +871,86 @@ values.
                 "metadata",
                 "images"
             ]
+        }
+    ]
+~~~
+
+The access requested for a given object when using these fields 
+is the cross-product of all fields of the object. That is to 
+say, the object represents a request for all `action` values listed within the object
+to be used at all `locations` values listed within the object for all `datatype`
+values listed within the object. In the example above,
+the RC is requesting both `read` and `write` access to two RS locations for
+both `metadata` and `images` within the API.
+
+To request a different combination of access, 
+such requesting one `action` against one `location` 
+and a different `action` against a different `location`, the 
+RC can include multiple separate objects in the `resources` array.
+The following non-normative example uses the same fictitious `photo-api`
+type definition to request a single access token with more specifically
+targeted access rights by using two discrete objects within the request.
+
+~~~
+    "resources": [
+        {
+            "type": "photo-api",
+            "actions": [
+                "read"
+            ],
+            "locations": [
+                "https://server.example.net/"
+            ],
+            "datatypes": [
+                "images"
+            ]
+        },
+        {
+            "type": "photo-api",
+            "actions": [
+                "write"
+            ],
+            "locations": [
+                "https://resource.local/other"
+            ],
+            "datatypes": [
+                "metadata"
+            ]
+        }
+    ]
+~~~
+
+The access requested here is for `read` access to `images` on one server
+while simultaneously requesting `write` access for `metadata` on a different
+server, but importantly without requesting `write` access to `images` on the
+first server.
+
+It is anticipated that API designers will use a combination
+of common fields defined in this specification as well as
+fields specific to the API itself. The following non-normative 
+example shows the use of both common and API-specific fields as 
+part of two different access `type` values. 
```

Comment:
here geolocation

Reply:
OK, and there's also the `currency` from the second API that we can point out. I didn't do that before because I didn't want people looking for those field definitions, but I think I was overthinking it. I'll update to address that.